editor/plugins/skeleton_editor_plugin.cpp

```diff
@@ -30,13 +30,326 @@
 
 #include "skeleton_editor_plugin.h"
 
+#include "core/io/resource_saver.h"
+#include "editor/editor_file_dialog.h"
+#include "editor/editor_properties.h"
+#include "editor/editor_scale.h"
+#include "editor/plugins/animation_player_editor_plugin.h"
 #include "scene/3d/collision_shape.h"
+#include "scene/3d/mesh_instance.h"
 #include "scene/3d/physics_body.h"
 #include "scene/3d/physics_joint.h"
 #include "scene/resources/capsule_shape.h"
 #include "scene/resources/sphere_shape.h"
 #include "spatial_editor_plugin.h"
 
+void BoneTransformEditor::create_editors() {
+	const Color section_color = get_color("prop_subsection", "Editor");
+
+	section = memnew(EditorInspectorSection);
+	section->setup("trf_properties", label, skeleton, section_color, true);
+	add_child(section);
+
+	key_button = memnew(Button);
+	key_button->set_text(TTR("Key Transform"));
+	key_button->set_visible(keyable);
+	key_button->set_icon(get_icon("Key", "EditorIcons"));
+	key_button->set_flat(true);
+	section->get_vbox()->add_child(key_button);
+
+	enabled_checkbox = memnew(CheckBox(TTR("Pose Enabled")));
+	enabled_checkbox->set_flat(true);
+	enabled_checkbox->set_visible(toggle_enabled);
+	section->get_vbox()->add_child(enabled_checkbox);
+
+	Label *l1 = memnew(Label(TTR("Translation")));
+	section->get_vbox()->add_child(l1);
+
+	translation_grid = memnew(GridContainer());
+	translation_grid->set_columns(3);
+	section->get_vbox()->add_child(translation_grid);
+
+	Label *l2 = memnew(Label(TTR("Rotation Degrees")));
+	section->get_vbox()->add_child(l2);
+
+	rotation_grid = memnew(GridContainer());
+	rotation_grid->set_columns(3);
+	section->get_vbox()->add_child(rotation_grid);
+
+	Label *l3 = memnew(Label(TTR("Scale")));
+	section->get_vbox()->add_child(l3);
+
+	scale_grid = memnew(GridContainer());
+	scale_grid->set_columns(3);
+	section->get_vbox()->add_child(scale_grid);
+
+	Label *l4 = memnew(Label(TTR("Transform")));
+	section->get_vbox()->add_child(l4);
+
+	transform_grid = memnew(GridContainer());
+	transform_grid->set_columns(3);
+	section->get_vbox()->add_child(transform_grid);
+
+	static const char *desc[12] = { "x", "y", "z", "x", "y", "z", "x", "y", "z", "x", "y", "z" };
+
+	for (int i = 0; i < 3; ++i) {
+		translation[i] = memnew(EditorSpinSlider());
+		translation[i]->set_label(desc[i]);
+		setup_spinner(translation[i], false);
+		translation[i]->set_step(0.1);
+		translation_grid->add_child(translation[i]);
+
+		rotation[i] = memnew(EditorSpinSlider());
+		rotation[i]->set_label(desc[i]);
+		rotation[i]->set_flat(true);
+		setup_spinner(rotation[i], false);
+		rotation[i]->set_step(0.1);
```

Comment:
FYI, if no step is defined, it will default to the `interface/inspector/default_float_step` editor setting, which is `0.001` by default. So indeed, not specifying the step is the right approach here if it's not meant to be particularly limited.

Reply:
The 12 stands for the number of floats in a Transform. 3x3 (scale and rotation) + 1x3 (origin).

Reply:
The translation range is (-inf, inf).
The rotation range is multiples of 0 to 360 that include the negatives.
The scale is mostly around -1 to 1. But it can be any number.